services/catalog/rs_server_catalog/main.py

```diff
@@ -68,15 +68,16 @@ def extract_openapi_specification():
     )
     openapi_spec_paths = openapi_spec["paths"]
     for key, _ in openapi_spec_paths.items():
-        new_key = "/catalog/{owner_id}" + key
+        new_key = f"/catalog{key}" if key == "/search" else "/catalog/{owner_id}" + key
         openapi_spec_paths[new_key] = openapi_spec_paths.pop(key)
         endpoint = openapi_spec_paths[new_key]
         for method_key, _ in endpoint.items():
             method = endpoint[method_key]
-            if "parameters" in method.keys():
-                method["parameters"] = add_parameter_owner_id(method["parameters"])
-            else:
-                method["parameters"] = add_parameter_owner_id([])
+            if new_key != "/catalog/search":
+                if "parameters" in method.keys():
```

Comment:
Instead of whole if statement, you can replace with
method["parameters"] = add_parameter_owner_id(method.get("parameters"), [])